src/main/java/com/connexta/store/controllers/StoreController.java

```diff
@@ -69,19 +72,30 @@
     }
 
     final Long fileSize = file.getSize();
-    // TODO validate that fileSize is (0 GB, 10 GB]
+    if (fileSize > MAX_FILE_BYTES) {
+      throw new CreateDatasetException(
+          BAD_REQUEST,
+          String.format(
+              "File size is %d bytes. File size cannot be greater than %d bytes",
+              fileSize, MAX_FILE_BYTES));
+    }
 
     final String mediaType = file.getContentType();
-    // TODO verify that mediaType is not blank and is a valid Content Type
+    if (mediaType == null) {
```

Comment:
✏️  If the mediaType is not null could it still be invalid? If so, I think it should be validated here, or we should keep that TODO.

Reply:
AFAIK we except any media type. At this stage in the application, the File is just a stream of bits. We store it and ask transform to interpret it. If transform can interpret the File enough to know if it can be promoted, then all is well.

Reply:
I'll go ahead and add a test for empty or blank strings, but I think they are unlikely. These kinds of things are either null or well-formed. Usually.